### PR TITLE
908 add cuda to ci serialize

### PR DESCRIPTION
**Pull request by pnstickne**
Fixes the upfront NVCC / CUDA 10.1 template issues. Compilation still fails much later where it looks like the checkpoint is not picking up on a serialization method.

### CODE REVIEW

File: src/vt/messaging/message/message_serialize.h [hunk 1]
```diff
@@ -53,11 +53,37 @@
...
 // causing templated instantiation (which will fail).
 #ifndef vt_quirked_serialize_method_detection
 
+using SerializeSizerType = ::checkpoint::Sizer;
+
+template <class T>
+struct has_any_serialize_member_t
+{
+  template <typename U>
+  static auto test(U* u) -> decltype(
```
Comment (pnstickne) [inline src/vt/messaging/message/message_serialize.h:80]: NVCC is 'okay' with decltype used like this, even when the member is deleted.

File: src/vt/messaging/message/message_serialize.h [hunk 2]
```diff
@@ -77,7 +104,14 @@ struct has_own_serialize_member_t<U,
 template <typename T>
 static constexpr auto const has_own_serialize =
   ::checkpoint::SerializableTraits<T>::has_serialize_noninstrusive
-  or has_own_serialize_member_t<T>::value;
+  // NVCC (CUDA 10.1) does use lazy evaluation with 'and' substituted here.
+  // The check for has_any.. is also required for NVCC which will
+  // fail compilation at the decltype usage in has_own.., although
+  // correctly not-substitute in has_any..
+  or cxx14_conjunction<
+    has_any_serialize_member_t<T>,
```
Comment (pnstickne) [inline src/vt/messaging/message/message_serialize.h:112]: `has_any.. and has_own..` still fails in NVCC even though it should short-circuit on `has_any..` failing.

File: src/vt/messaging/active.impl.h [hunk 3]
```diff
@@ -55,7 +55,7 @@
 
 namespace vt { namespace messaging {
 
-constexpr ByteType msgsize_not_specified = -1;
+constexpr ByteType msgsize_not_specified = static_cast<ByteType>(-1);
```
Comment (pnstickne) [inline src/vt/messaging/active.impl.h:58]: Add note about signage change?

File: src/vt/messaging/message/message_serialize.h (hunk 2 above)
Comment (PhilMiller) [reply src/vt/messaging/message/message_serialize.h:112]: Try using `&&` and `||` instead of the keywords.

**Comment by pnstickne** [comment]
Broke GitHub PR somehow 🤷. Re-created as https://github.com/DARMA-tasking/vt/pull/919.

**Pull request closed without merging**